base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs

```diff
@@ -323,12 +323,21 @@ where
             let sig = response.signature;
             if let Some(unconfirmed_tx) = batch_signatures.get(&sig) {
                 if response.location == TxLocation::Mined {
-                    mined.push((
-                        (*unconfirmed_tx).clone(),
-                        response.block_height,
-                        response.block_hash.unwrap(),
-                        response.confirmations,
-                    ));
+                    if response.block_hash.is_some() {
```

Comment:
I had to look into this a bit. I was concerned is_some() protects us from passing an empty option, but I was curious what would happen if our block hash was made of just an empty Vec<u8>. Was there a chance we could have created an empty vector and as a result diesel saved it in the db as null? 
It turns out this isn't a concern as diesel will use a specific null binary for an empty vector. Our example DB contains no data in the field so this must not be our current case, and likely not a concern here.